src/main/java/seedu/address/logic/parser/FindCommandParser.java

```diff
@@ -19,15 +26,54 @@
      * @throws ParseException if the user input does not conform the expected format
      */
     public FindCommand parse(String args) throws ParseException {
-        String trimmedArgs = args.trim();
-        if (trimmedArgs.isEmpty()) {
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_PHONE,
+                        PREFIX_EMAIL, PREFIX_ADDRESS, PREFIX_TAG);
+
+        ContactContainsFieldsPredicate findPredicate = new ContactContainsFieldsPredicate();
+        if (argMultimap.getValue(PREFIX_NAME).isPresent()) {
+            String trimmedNameKeywords = parseNonTagPredicateField(argMultimap.getValue(PREFIX_NAME).get());
+            findPredicate.setNameKeywords(Arrays.asList(trimmedNameKeywords.split("\\s+")));
+        }
+        if (argMultimap.getValue(PREFIX_PHONE).isPresent()) {
+            String trimmedPhoneKeyword = parseNonTagPredicateField(argMultimap.getValue(PREFIX_PHONE).get());
+            findPredicate.setPhoneKeyword(trimmedPhoneKeyword);
+        }
+        if (argMultimap.getValue(PREFIX_EMAIL).isPresent()) {
+            String trimmedEmailKeyword = parseNonTagPredicateField(argMultimap.getValue(PREFIX_EMAIL).get());
+            findPredicate.setEmailKeyword(trimmedEmailKeyword);
+        }
+        if (argMultimap.getValue(PREFIX_ADDRESS).isPresent()) {
+            String trimmedAddressKeyword = parseNonTagPredicateField(argMultimap.getValue(PREFIX_ADDRESS).get());
+            findPredicate.setAddressKeyword(trimmedAddressKeyword);
+        }
+        findPredicate.setTagKeywords(parseTagPredicateFields(argMultimap.getAllValues(PREFIX_TAG)));
+
+        if (findPredicate.isEmptyPredicate()) {
             throw new ParseException(
                     String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
         }
 
-        String[] nameKeywords = trimmedArgs.split("\\s+");
+        return new FindCommand(findPredicate);
+    }
 
-        return new FindCommand(new NameContainsKeywordsPredicate(Arrays.asList(nameKeywords)));
+    /** Returns a trimmed predicateField. If the field is not valid, throws a ParseException instead. */
```

Comment:
for consistency across the entire codebase, i think it'll be better if you have your documentation written like this:

```
/**
 * Returns a trimmed predicateField. If...
 */
```